app/api/v1/endpoints/prediction.py

```
@@ -1,45 +1,37 @@
-    # URL에서 이미지를 다운로드
+@router.post("/vision/predict", response_model=List[PredictionResult])
+async def predict_image_custom_vision(request: ImagePredictionRequest):
     try:
-        response = requests.get(request.image_url)
-        response.raise_for_status()
-    except requests.exceptions.HTTPError as err:
-        raise HTTPException(status_code=400, detail="Error fetching image: " + str(err)) from err
+        predictions = await predict_pet_disease_custom_vision(request.image_url, request.pet_type)
+        return predictions
+    except ValueError as e:  # pet_type 오류 처리
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)